### PR TITLE
Initial implementation of Hyper Novice

### PR DESCRIPTION
**Pull request by Lemongrass3110**
* **Addressed Issue(s)**: None

* **Server Mode**: Renewal

* **Description of Pull Request**: 
Taken from #7024

### CODE REVIEW

File: src/map/battle.cpp [hunk 1]
```diff
@@ -1569,6 +1575,11 @@ int64 battle_calc_damage(struct block_list *src,struct block_list *bl,struct Dam
 			damage *= 4;
 #endif
 
+		if (tsc->getSCE(SC_GROUNDGRAVITY) && flag&(BF_MAGIC|BF_WEAPON) && !status_bl_has_mode(bl, MD_STATUSIMMUNE))
```
Comment (Lemongrass3110) [inline src/map/battle.cpp:1578]:
```suggestion
		if (tsc->getSCE(SC_GROUNDGRAVITY) && flag&(BF_MAGIC|BF_WEAPON))
```
Do we really need to check this again? It cannot be started on bosses anyway.

File: src/map/battle.cpp [hunk 2]
```diff
@@ -1569,6 +1575,11 @@ int64 battle_calc_damage(struct block_list *src,struct block_list *bl,struct Dam
...
+			damage += damage * 15 / 100;
+		if (tsc->getSCE(SC_SHIELDCHAINRUSH) && !status_bl_has_mode(bl, MD_STATUSIMMUNE))
```
Comment (Lemongrass3110) [inline src/map/battle.cpp:1580]:
```suggestion
		if (tsc->getSCE(SC_SHIELDCHAINRUSH))
```
Do we really need to check this again? It cannot be started on bosses anyway.

File: src/map/skill.cpp [hunk 3]
```diff
@@ -12792,6 +12862,11 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 		}
 		break;
 
+	case HN_HELLS_DRIVE:
```
Comment (Lemongrass3110) [inline src/map/skill.cpp:12865]: Should this really be in skill_castend_nodamage_id?
Or should it be moved to skill_castend_damage_id?

Comment (aleos89) [reply src/map/skill.cpp:12865]: Agreed, this can be moved to the generic splash skill list!

File: src/map/battle.cpp (hunk 1 above)
Comment (aleos89) [reply src/map/battle.cpp:1578]: Agreed, the status has `BossResist` so it should not have applied to a boss type monster.

File: src/map/battle.cpp (hunk 2 above)
Comment (aleos89) [reply src/map/battle.cpp:1580]: Agreed, the status has `BossResist` so it should not have applied to a boss type monster.

**Pull request merged**